src/backend/distributed/planner/multi_join_order.c

```diff
@@ -188,60 +176,11 @@ FixedJoinOrderList(FromExpr *fromExpr, List *tableEntryList)
 			joinClauseList = list_concat(joinClauseList, joinWhereClauseList);
 		}
 
-		/* get the sorted list of shards to check broadcast/local join possibility */
-		candidateShardList = LoadShardIntervalList(nextTable->relationId);
-
 		/* find the best join rule type */
 		nextJoinNode = EvaluateJoinRules(joinedTableList, currentJoinNode,
-										 nextTable, candidateShardList,
```

Comment:
@mtuncer and @marcocitus we're doing a major code refactoring for join order planning. In summary:
* For broadcast joins, rely only on reference tables (e.g., get rid of `large_table_shard_count`) 
* For local joins, rely on `CoPartitionedTables()`

The implication of those changes in `FixedJoinOrder()` seems substantial. In general, it simplifies the code a lot (as expected) and all changes make sense to me. Feel free to have a look and comment on the changes.